lib/src/main/java/com/tubitv/media/fsm/concrete/MoviePlayingState.java

```diff
@@ -53,9 +54,14 @@ private void stopAdandPlayerMovie(PlayerUIController controller, PlayerAdLogicCo
         SimpleExoPlayer adPlayer = controller.getAdPlayer();
         SimpleExoPlayer moviePlayer = controller.getContentPlayer();
 
+        boolean shouldReprepareForSinglePlayer = PlayerDeviceUtils.useSinglePlayer() && controller.isPlayingAds;
+
         //first remove the AdPlayer's listener and pause the player
-        adPlayer.removeListener(componentController.getAdPlayingMonitor());
-        adPlayer.setPlayWhenReady(false);
+        adPlayer.removeAnalyticsListener(componentController.getAdPlayingMonitor());
+
+        if (shouldReprepareForSinglePlayer) {
+            adPlayer.setPlayWhenReady(false);
+        }
```

Comment:
whether using singleplayer or double player, we need to set the adplayer.setPlayWhenReady(false);